src/super.c

```diff
@@ -387,10 +418,59 @@ static int assign_random_id(struct scoutfs_sb_info *sbi)
 	return 0;
 }
 
+/*
+ * Ensure superblock copies in metadata and data block devices are valid, and
+ * fill in in-memory superblock if so.
+ */
+static int scoutfs_read_supers(struct super_block *sb)
+{
+	struct scoutfs_super_block *meta_super, *data_super;
+	struct scoutfs_sb_info *sbi = SCOUTFS_SB(sb);
+	int ret = 0;
+
+	meta_super = scoutfs_read_super_from_bdev(sb, sbi->meta_bdev);
+	if (IS_ERR(meta_super)) {
+		scoutfs_err(sb, "could not get meta_super");
```

Comment:
Let's print the errnos in all these error cases.   And if we initialized meta and data_super to null we could have all the cases goto out and free and return.